File: leetcode/top-interview-questions-easy/strings/1. reverse_string.js
/**
 * @param {character[]} s
 * @return {void} Do not return anything, modify s in-place instead.
 */
var reverseString = function (s) {
  var len = s.length;
  for (let i = 0; i < len / 2; i++) {
    let lastIndex = len - i - 1;
    let first = s[i];
    let last = s[lastIndex];
    s[i] = last;
    s[lastIndex] = first;
  }
};

File: leetcode/top-interview-questions-easy/strings/1. reverse_string.py
class Solution(object):
    def reverseString(self, s):
        leng = len(s)
        for i in range(0, leng / 2):
            lastIndex = leng - i - 1
            first = s[i]
            last = s[lastIndex]
            s[i] = last
            s[lastIndex] = first